front/lib/api/assistant/conversation.ts

```diff
@@ -1632,17 +1633,24 @@ export async function postNewContentFragment(
 
   const messageId = generateModelSId();
 
-  const sourceUrl = isSupportedUploadableContentFragmentType(contentType)
-    ? fileAttachmentLocation({
-        workspaceId: owner.sId,
-        conversationId: conversation.sId,
-        messageId,
-        contentFormat: "raw",
-      }).downloadUrl
-    : url;
-
-  // TODO(2024-06-27 flav) Consider resizing images.
+  let sourceUrl: string | null = null;
+  if (url && url.startsWith(FileResource.baseFileUrl(owner))) {
+    // Use the provided URL if it is an internal file path.
+    sourceUrl = url;
+  } else if (isSupportedUploadableContentFragmentType(contentType)) {
+    // Deprecated, for supported content types, create a file path and use its download URL.
+    sourceUrl = fileAttachmentLocation({
+      workspaceId: owner.sId,
+      conversationId: conversation.sId,
+      messageId,
+      contentFormat: "raw",
+    }).downloadUrl;
+  } else {
+    // Otherwise, use the provided URL (e.g., a Slack thread).
+    sourceUrl = url;
+  }
```

Comment:
Not sure what's different than previously?
Basically if we not in `isSupportedUploadableContentFragmentType(contentType)` we set `sourceUrl = url`, right?

Reply:
The difference is that know the client passes an url that points toward our own infrastructure, if this is the case, we use it as the url, otherwise we fallback to the previous logic.

Reply:
Yes but unless I'm missing something this is equivalent to:
```typescript
if (isSupportedUploadableContentFragmentType(contentType)) {
    // Deprecated, for supported content types, create a file path and use its download URL.
    sourceUrl = fileAttachmentLocation({
      workspaceId: owner.sId,
      conversationId: conversation.sId,
      messageId,
      contentFormat: "raw",
    }).downloadUrl;
  } else {
    sourceUrl = url;
  }
```